### ISSUE TITLE
aws_vpn_connection - terraform plan output is different on different machines

### ISSUE THREAD

**Issue by gg-chrisd** [hashicorp/terraform-provider-aws#18327]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

Terraform v0.13.6
+ provider registry.terraform.io/-/aws v3.33.0
+ provider registry.terraform.io/hashicorp/aws v2.14.0

### Affected Resource(s)

aws_vpn_connection

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
resource aws_vpn_gateway vgw {
  vpc_id = aws_vpc.vpc.id
}

resource aws_customer_gateway hq {
  bgp_asn    = 65000
  ip_address = "xxx.xxx.xxx.xxx"
  type       = "ipsec.1"
}

resource aws_vpn_connection vpn {
  vpn_gateway_id          = aws_vpn_gateway.vgw.id
  customer_gateway_id = aws_customer_gateway.hq.id
  type                            = aws_customer_gateway.hq.type
  static_routes_only       = true
}
```

### Debug Output

For the working Ubuntu Linux (WSL) machine, no changes are required.

For the non-working Mac machine, see the [gist](https://gist.github.com/gg-chrisd/72c264c3e4d21a21277c28f0921eb1af)

### Expected Behavior

There should be No changes required for both machines.

### Actual Behavior

See the gist, it wants to alter/remove the algorithms associated with the VPN connection

### Steps to Reproduce

1. Pull the repository to a fresh machine
2. Connect the AWS CLI
3. Run `terraform plan` and check the output

Things we've tried:
- removing the repo and re-pulling.
- Running `terraform state rm aws_vpn_connection.vpn` and re-importing

Still the same results.

### Important Factoids

Using remote S3 state.
Machines tested on were 1 Ubuntu Linux (working as expected) and 1 Mac (Displaying the issue)

**Comment by gdavison** [comment]
Hi @gg-chrisd, thank you for submitting this issue. Can you please confirm that you're running the same version of Terraform and the same version of the AWS Provider in both cases?

In both the WSL and macOS cases, are you deploying into a fresh environment, or are you applying first with one and then updating on the other?

Can you also include `DEBUG`-level logs for both WSL and macOS? There are instructions for enabling logging at https://www.terraform.io/docs/internals/debugging.html. Debug logs tend to be large, so it is best to use a Gist to include them.

**Comment by gg-chrisd** [comment]
Hi @gdavison, thanks for getting back to me.

I've found the issue, previously we had the provider set to `2.14.0` but after upgrading to 0.13 I removed this from the provider block and assumed this would then update the provider to the latest following a `terraform init`, but after looking at the debug logs I could see `2.14.0` still being used.

So, I looked in the `.terraform` folder on the WSL machine, I noticed that I had a `-/aws/3.33.0 folder` but a `hashicorp/aws/2.14.0` folder. After removing the latter and performing a `terraform init`, then `terraform plan` I'm now seeing that it wants to update the ciphers used on the VPN resource in the same manner as the Mac (Fresh install). However, this fails with a 400 error, so for the time-being I've added a `version= "~> 2.14.0"` back in to the `aws provider` block.

So, I think we can close this issue and I'll take a look at the 400 error separately.

Thanks
Chris
